osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js

```diff
@@ -1,56 +1,111 @@
-inputElementSouth.addEventListener('change', updateMapBoundingBox);
+(function(){
+    var ExcerptManager = function(locationFilter, inputElementsNewBoundingBox, selectElementExistingExcerpts, formElementPartsSwitcher) {
+        this.locationFilter = locationFilter;
+        this.inputElementsNewBoundingBox = inputElementsNewBoundingBox;
+        this.selectElementExistingExcerpts = selectElementExistingExcerpts;
+        this.formElementPartsSwitcher = formElementPartsSwitcher;
+
+
+        /**
+         * Synchronize coordinates in input fields to excerpt on map
+         */
+        this.updateInputElementsBoundingBox = function() {
+            var locationFilterBounds = this.locationFilter.getBounds();
+            this.inputElementsNewBoundingBox.inputElementNorth.value = locationFilterBounds._northEast.lat;
+            this.inputElementsNewBoundingBox.inputElementWest.value = locationFilterBounds._southWest.lng;
+            this.inputElementsNewBoundingBox.inputElementEast.value = locationFilterBounds._northEast.lng;
+            this.inputElementsNewBoundingBox.inputElementSouth.value = locationFilterBounds._southWest.lat;
+        };
+
+        /**
+         * Synchronize excerpt on map to coordinates in input fields
+         */
+        this.updateMapExcerpt = function() {
+            var locationFilterBounds = this.locationFilter.getBounds();
+            locationFilterBounds._northEast.lat = this.inputElementsNewBoundingBox.inputElementNorth.value;
+            locationFilterBounds._southWest.lng = this.inputElementsNewBoundingBox.inputElementWest.value;
+            locationFilterBounds._northEast.lng = this.inputElementsNewBoundingBox.inputElementEast.value;
+            locationFilterBounds._southWest.lat = this.inputElementsNewBoundingBox.inputElementSouth.value;
+            this.locationFilter.setBounds(locationFilterBounds);
+        };
+
+        /**
+         * if the shown excerpt on the map changes and the change was not made by selecting an existing excerpt in the list
+         * -> a user changed the excerpt, so change the form part to 'new-excerpt'
+         */
+        this.userChangeExcerptOnMapShowNewExcerptPart = function() {
+            var locationFilterBounds = this.locationFilter.getBounds();
+            var select = this.selectElementExistingExcerpts;
+            if(!(selectElementExistingExcerpts.value != "" &&
+                    select.querySelector('option[value="'+select.value+'"]').getAttribute('data-north') == locationFilterBounds._northEast.lat &&
+                    select.querySelector('option[value="'+select.value+'"]').getAttribute('data-west') == locationFilterBounds._southWest.lng &&
+                    select.querySelector('option[value="'+select.value+'"]').getAttribute('data-east') == locationFilterBounds._northEast.lng &&
+                    select.querySelector('option[value="'+select.value+'"]').getAttribute('data-south') == locationFilterBounds._southWest.lat)) {
```

Comment:
consider extracting `select.querySelector('option[value="'+select.value+'"]').` as a local variable or `select.querySelector('option[value="'+select.value+'"]').getAttribute('data-' + direction)` as a function of `direction`